src/rules/typedefRule.ts

```diff
@@ -105,7 +121,17 @@ class TypedefWalker extends Lint.RuleWalker {
 
     public visitPropertyDeclaration(node: ts.PropertyDeclaration) {
         const optionName = "member-variable-declaration";
-        this.checkTypeAnnotation(optionName, node.name.getEnd(), node.type, node.name);
+        let performCheck = true;
+
+        // If this is an arrow function, it doesn't need to have a typedef on the property declaration
+        // as the typedefs can be on the function's parameters instead
+        if (!node.type && node.initializer.kind === ts.SyntaxKind.ArrowFunction) {
+            performCheck = false;
+        }
```

Comment:
I did this slightly differently as it read easier to me: `const performCheck = !(node.initializer.kind === ts.SyntaxKind.ArrowFunction && node.type == null);`